test/src/org/omegat/gui/glossary/GlossarySearcherTest.java

```diff
@@ -0,0 +1,272 @@
+        assertEquals(translationText, result.get(0).getLocText());
+    }
+
+    @Test
+    public void testGlossarySearcherJapanese2() {
+        String sourceText = "\u5834\u6240";
+        String commentText = "comment";
+        Language language = new Language("ja");
+        setupProject(language);
+        ITokenizer tok = new LuceneJapaneseTokenizer();
+        List<GlossaryEntry> entries = new ArrayList<>();
+        entries.add(new GlossaryEntry("\u5857\u5E03", "wrong", commentText, true, "origin"));
+        List<GlossaryEntry> result = glossarySearcherCommon(sourceText, tok, language, entries);
+        assertEquals(0, result.size());
+    }
+
+    @Test
+    public void testGlossarySearcherJapaneseLongText() {
+        Language language = new Language("ja");
+        setupProject(language);
+        ITokenizer tok = new LuceneJapaneseTokenizer();
+        List<GlossaryEntry> entries = new ArrayList<>();
+        entries.add(new GlossaryEntry("\u307E\u3050\u308D", "tuna", "", true, ""));
+        entries.add(new GlossaryEntry("\u7FFB\u8A33", "translation", "", true, ""));
+        entries.add(new GlossaryEntry("\u591A\u8A00\u8A9E", "multi-languages", "", true, ""));
+        entries.add(new GlossaryEntry("\u5730\u57DF\u5316", "localization", "", true, ""));
+        String sourceText = "OmegaT\u306E\u30E6\u30FC\u30B6\u30FC\u30A4\u30F3\u30BF\u30FC\u30D5\u30A7\u30FC\u30B9\u3084\u30D8\u30EB\u30D7\u30C6\u30AD\u30B9\u30C8\u3092\u3001\u3055\u307E\u3056\u307E\u306A\u8A00\u8A9E\u3078\u7FFB\u8A33\u3057\u3066\u304F\u3060\u3055\u3063\u305F\u65B9\u3005\u306B\u611F\u8B1D\u3057\u307E\u3059\u3002\u305D\u3057\u3066\u3001\u7FFB\u8A33\u304C\u306A\u3055\u308C\u3066\u3044\u306A\u3044\u8A00\u8A9E\u304C\u307E\u3060\u6570\u5343\u6B8B\u3063\u3066\u3044\u307E\u3059\uFF01OmegaT \u306E\u591A\u8A00\u8A9E\u3078\u306E\u5730\u57DF\u5316\u306F\u3001\u6301\u7D9A\u7684\u306A\u4F5C\u696D\u3067\u3082\u3042\u308A\u307E\u3059\u3002\u306A\u305C\u306A\u3089\u3001\u65B0\u3057\u3044\u6A5F\u80FD\u304C\u7D76\u3048\u305A\u8FFD\u52A0\u3055\u308C\u3066\u3044\u308B\u304B\u3089\u3067\u3059\u3002OmegaT\u306E\u30ED\u30FC\u30AB\u30E9\u30A4\u30BA/\u7FFB\u8A33\u306B\u95A2\u3059\u308B\u8A73\u7D30\u306B\u3064\u3044\u3066\u306F\u3001OmegaT\u30ED\u30FC\u30AB\u30EA\u30BC\u30FC\u30B7\u30E7\u30F3\u30B3\u30FC\u30C7\u30A3\u30CD\u30FC\u30BF\u30FC\u306B\u304A\u554F\u3044\u5408\u308F\u305B\u304F\u3060\u3055\u3044\u3002";
+        List<GlossaryEntry> result = glossarySearcherCommon(sourceText, tok, language, entries);
+        assertEquals(3, result.size());
+    }
+
+    private void setupProject(Language language) {
+        // setup project
+        final ProjectProperties props = new ProjectProperties() {
+            public Language getSourceLanguage() {
+                return language;
+            }
+
+            public Language getTargetLanguage() {
+                return new Language("pl");
+            }
+        };
+
+        Core.setProject(new IProject() {
```

Comment:
You could use an anonymous subclass of `NotLoadedProject` to avoid having to define a lot of the `IProject` methods; see e.g. `ExternalTMFactoryTest`.

Reply:
isCjkMatch expect `isProjectLoaded()`. Does `NotLoadedproject` work as stub? 

```
  protected static boolean isCjkMatch(String fullText, String term) {
        // This is a CJK word and our source language is not space-delimited, so include if
        // word appears anywhere in source string.
        IProject project = Core.getProject();
        return project.isProjectLoaded() && ...
```